### ISSUE TITLE
Indexer Crashing

### ISSUE THREAD

**Issue by cwils1689** [FaithLife-Community/LogosLinuxInstaller#167]
Hello everyone. First, thanks for this project. It's a blessing! 

I have a brand new System76 machine running Pop OS. The software is running pretty well. The only issue I've run into is my library will not index. When I try to manually run indexing it crashes Logos. If I try running the LogosIndexer.exe file directly it gives me an "unhandled exception" error.  Any advice? Also, I am not a Linux ninja, still learning the basics but enjoying it.

Thanks for any help!

**Comment by thw26** [comment]
Hello! Thank you for opening this issue.

Are you able to provide the full error log text?

**Comment by cwils1689** [comment]
Sure thing. 
`Unhandled exception: page fault on read access to 0x0000000c in 64-bit code (0x0000000000a35dc8).
Register dump:
 rip:0000000000a35dc8 rsp:000000000029d230 rbp:000000000029f030 eflags:00010206 (  R- --  I   - -P- )
 rax:000000000008e800 rbx:000000000029d280 rcx:0000000000000130 rdx:0000000000000000
 rsi:000000000029ef00 rdi:000000000029dc80  r8:00000000000003c9  r9:0000000019930520 r10:0000000000000005
 r11:000000000029d5a0 r12:0000000000000000 r13:000000000029d3a8 r14:000000000029fea0 r15:000000000029dde8
Stack dump:
0x000000000029d230:  000000000029dde8 000000000029fea0
0x000000000029d240:  0000000000b71380 000000007b070687
0x000000000029d250:  000000000029f0d0 0000000000c603f3
0x000000000029d260:  0000000000000004 000000000029d530
0x000000000029d270:  000000000029ef00 0000000000e95ee0
0x000000000029d280:  0000000000000000 0000000000b6fea0
0x000000000029d290:  0000000000000000 0000000000b80cb3
0x000000000029d2a0:  0000000000000000 0000000000e95ee0
0x000000000029d2b0:  000000000029dc80 0000000000b6fde1
0x000000000029d2c0:  000000000029d3a8 000000000029ef00
0x000000000029d2d0:  000000000029dc80 000000000029d530
0x000000000029d2e0:  000000000029ea00 0000000000b72390
Backtrace:
=>0 0x0000000000a35dc8 EntryPoint+0xffffffffffffffff() in coreclr (0x000000000029f030)
  1 0x0000000000c603f3 EntryPoint+0xf1e92() in coreclr (0x000000000029f030)
  2 0x0000000000b80cb3 EntryPoint+0x12752() in coreclr (0x000000000029f030)
  3 0x0000000000b72390 EntryPoint+0x3e2f() in coreclr (0x000000000029d530)
  4 0x0000000000b71495 EntryPoint+0x2f34() in coreclr (0x000000000029d530)
  5 0x000000007bc525a5 EntryPoint+0xffffffffffffffff() in ntdll (0x000000000029d530)
  6 0x0000000000a14d25 EntryPoint+0xffffffffffffffff() in coreclr (0x000000000029f3b1)
0x0000000000a35dc8 EntryPoint+0xffffffffffffffff in coreclr: movl	0x000000000000000c(%rdx),%eax
Modules:
Module	Address					Debug info	Name (36 modules)
PE	          890000-          8f4000	Deferred        hostpolicy
PE	          a10000-          ef5000	Export          coreclr
PE	        7b000000-        7b3fd000	Deferred        kernelbase
PE	        7b600000-        7b969000	Deferred        kernel32
PE	        7bc00000-        7bf3c000	Dwarf           ntdll
PE	       140000000-       140030000	Deferred        logosindexer
PE	       180000000-       180059000	Deferred        hostfxr
PE	       1c8db0000-       1c9097000	Deferred        msvcrt
PE	       1cd860000-       1cd863000	Deferred        api-ms-win-crt-utility-l1-1-0
PE	       1d97a0000-       1d97ae000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	       21a7e0000-       21a9ce000	Deferred        setupapi
PE	       231ae0000-       231d60000	Deferred        rpcrt4
PE	       23d820000-       23dfc2000	Deferred        user32
PE	       241850000-       241853000	Deferred        api-ms-win-crt-environment-l1-1-0
PE	       262250000-       262254000	Deferred        api-ms-win-crt-runtime-l1-1-0
PE	       26b4c0000-       26bbb9000	Deferred        gdi32
PE	       2739c0000-       273ec9000	Deferred        oleaut32
PE	       28ba60000-       28ba63000	Deferred        api-ms-win-crt-time-l1-1-0
PE	       2e3540000-       2e364f000	Deferred        shlwapi
PE	       2e8f10000-       2e948f000	Deferred        ole32
PE	       2f1fa0000-       2f1fbe000	Deferred        version
PE	       30a2c0000-       30a2c4000	Deferred        api-ms-win-crt-stdio-l1-1-0
PE	       30c980000-       30c98f000	Deferred        api-ms-win-core-synch-l1-2-0
PE	       3126f0000-       312744000	Deferred        shcore
PE	       327020000-       3271a4000	Deferred        combase
PE	       32a700000-       32a7bc000	Deferred        sechost
PE	       330260000-       330387000	Deferred        advapi32
PE	       33ea00000-       33ea04000	Deferred        api-ms-win-crt-string-l1-1-0
PE	       344840000-       344843000	Deferred        api-ms-win-crt-filesystem-l1-1-0
PE	       350a30000-       350a34000	Deferred        api-ms-win-crt-convert-l1-1-0
PE	       355100000-       355103000	Deferred        api-ms-win-crt-locale-l1-1-0
PE	       360a80000-       360a85000	Deferred        api-ms-win-crt-math-l1-1-0
PE	       39b510000-       39b513000	Deferred        api-ms-win-crt-heap-l1-1-0
PE	       3af670000-       3af762000	Deferred        ucrtbase
PE	       3afd00000-       3afd62000	Deferred        imm32
PE	    7afe08220000-    7afe08af2000	Deferred        shell32
Threads:
process  tid      prio (all id:s are in hex)
00000038 services.exe
	0000003c    0
	00000040    0
	00000054    0
	00000058    0
	00000070    0
	00000084    0
	000000a0    0
	000000c0    0
	000000c4    0
	000000e8    0
	00000100    0
00000044 mscorsvw.exe
	00000048    0
	0000005c    0
	00000060    0
	00000064    0
0000004c explorer.exe
	00000050    0
	0000009c    0
	000000b8    0
00000068 svchost.exe
	0000006c    0
	00000074    0
	00000078    0
0000007c winedevice.exe
	00000080    0
	00000088    0
	0000008c    0
	00000090    0
00000094 plugplay.exe
	00000098    0
	000000a4    0
	000000a8    0
	000000ac    0
	000000d4    0
	000000d8    0
000000b0 winedevice.exe
	000000b4    0
	000000c8    0
	000000cc    0
	000000d0    0
	000000dc    0
000000e0 mscorsvw.exe
	000000e4    0
	000000ec    0
	000000f0    0
	000000f4    0
000000f8 rpcss.exe
	000000fc    0
	00000104    0
	00000108    0
	0000010c    0
	00000110    0
	00000114    0
	00000118    0
0000011c (D) Z:\home\clintwilson\LogosBible10\data\wine64_bottle\drive_c\users\clintwilson\AppData\Local\Logos\System\LogosIndexer.exe
	00000120    0 <==
	0000012c    0
System information:
    Wine build: wine-6.0.3 (Ubuntu 6.0.3~repack-1)
    Platform: x86_64
    Version: Windows 10
    Host system: Linux
    Host version: 6.9.3-76060903-generic`

**Comment by thw26** [comment]
Thank you. The code that comes before this is called the stack trace. Can you provide all the code? Furthermore, can you submit to a service like Pastebin for easy reading?

https://pastebin.com/

**Comment by cwils1689** [comment]
Okay. Thanks for that link. I'm not seeing any code before this. I'm afraid I'm not looking in the right place.

`Unhandled exception: page fault on read access to 0x0000000c in 64-bit code (0x0000000000a35dc8).
Register dump:
 rip:0000000000a35dc8 rsp:000000000029d230 rbp:000000000029f030 eflags:00010206 (  R- --  I   - -P- )
 rax:000000000008e800 rbx:000000000029d280 rcx:0000000000000130 rdx:0000000000000000
 rsi:000000000029ef00 rdi:000000000029dc80  r8:00000000000003c9  r9:0000000019930520 r10:0000000000000005
 r11:000000000029d5a0 r12:0000000000000000 r13:000000000029d3a8 r14:000000000029fea0 r15:000000000029dde8
Stack dump:
0x000000000029d230:  000000000029dde8 000000000029fea0
0x000000000029d240:  0000000000b71380 000000007b070687
0x000000000029d250:  000000000029f0d0 0000000000c603f3
0x000000000029d260:  0000000000000004 000000000029d530
0x000000000029d270:  000000000029ef00 0000000000e95ee0
0x000000000029d280:  0000000000000000 0000000000b6fea0
0x000000000029d290:  0000000000000000 0000000000b80cb3
0x000000000029d2a0:  0000000000000000 0000000000e95ee0
0x000000000029d2b0:  000000000029dc80 0000000000b6fde1
0x000000000029d2c0:  000000000029d3a8 000000000029ef00
0x000000000029d2d0:  000000000029dc80 000000000029d530
0x000000000029d2e0:  000000000029ea00 0000000000b72390
Backtrace:
=>0 0x0000000000a35dc8 EntryPoint+0xffffffffffffffff() in coreclr (0x000000000029f030)
  1 0x0000000000c603f3 EntryPoint+0xf1e92() in coreclr (0x000000000029f030)
  2 0x0000000000b80cb3 EntryPoint+0x12752() in coreclr (0x000000000029f030)
  3 0x0000000000b72390 EntryPoint+0x3e2f() in coreclr (0x000000000029d530)
  4 0x0000000000b71495 EntryPoint+0x2f34() in coreclr (0x000000000029d530)
  5 0x000000007bc525a5 EntryPoint+0xffffffffffffffff() in ntdll (0x000000000029d530)
  6 0x0000000000a14d25 EntryPoint+0xffffffffffffffff() in coreclr (0x000000000029f3b1)
0x0000000000a35dc8 EntryPoint+0xffffffffffffffff in coreclr: movl	0x000000000000000c(%rdx),%eax
Modules:
Module	Address					Debug info	Name (36 modules)
PE	          890000-          8f4000	Deferred        hostpolicy
PE	          a10000-          ef5000	Export          coreclr
PE	        7b000000-        7b3fd000	Deferred        kernelbase
PE	        7b600000-        7b969000	Deferred        kernel32
PE	        7bc00000-        7bf3c000	Dwarf           ntdll
PE	       140000000-       140030000	Deferred        logosindexer
PE	       180000000-       180059000	Deferred        hostfxr
PE	       1c8db0000-       1c9097000	Deferred        msvcrt
PE	       1cd860000-       1cd863000	Deferred        api-ms-win-crt-utility-l1-1-0
PE	       1d97a0000-       1d97ae000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	       21a7e0000-       21a9ce000	Deferred        setupapi
PE	       231ae0000-       231d60000	Deferred        rpcrt4
PE	       23d820000-       23dfc2000	Deferred        user32
PE	       241850000-       241853000	Deferred        api-ms-win-crt-environment-l1-1-0
PE	       262250000-       262254000	Deferred        api-ms-win-crt-runtime-l1-1-0
PE	       26b4c0000-       26bbb9000	Deferred        gdi32
PE	       2739c0000-       273ec9000	Deferred        oleaut32
PE	       28ba60000-       28ba63000	Deferred        api-ms-win-crt-time-l1-1-0
PE	       2e3540000-       2e364f000	Deferred        shlwapi
PE	       2e8f10000-       2e948f000	Deferred        ole32
PE	       2f1fa0000-       2f1fbe000	Deferred        version
PE	       30a2c0000-       30a2c4000	Deferred        api-ms-win-crt-stdio-l1-1-0
PE	       30c980000-       30c98f000	Deferred        api-ms-win-core-synch-l1-2-0
PE	       3126f0000-       312744000	Deferred        shcore
PE	       327020000-       3271a4000	Deferred        combase
PE	       32a700000-       32a7bc000	Deferred        sechost
PE	       330260000-       330387000	Deferred        advapi32
PE	       33ea00000-       33ea04000	Deferred        api-ms-win-crt-string-l1-1-0
PE	       344840000-       344843000	Deferred        api-ms-win-crt-filesystem-l1-1-0
PE	       350a30000-       350a34000	Deferred        api-ms-win-crt-convert-l1-1-0
PE	       355100000-       355103000	Deferred        api-ms-win-crt-locale-l1-1-0
PE	       360a80000-       360a85000	Deferred        api-ms-win-crt-math-l1-1-0
PE	       39b510000-       39b513000	Deferred        api-ms-win-crt-heap-l1-1-0
PE	       3af670000-       3af762000	Deferred        ucrtbase
PE	       3afd00000-       3afd62000	Deferred        imm32
PE	    7afe08220000-    7afe08af2000	Deferred        shell32
Threads:
process  tid      prio (all id:s are in hex)
00000038 services.exe
	0000003c    0
	00000040    0
	00000054    0
	00000058    0
	00000070    0
	00000084    0
	000000a0    0
	000000c0    0
	000000c4    0
	000000e8    0
	00000100    0
00000044 mscorsvw.exe
	00000048    0
	0000005c    0
	00000060    0
	00000064    0
0000004c explorer.exe
	00000050    0
	0000009c    0
	000000b8    0
00000068 svchost.exe
	0000006c    0
	00000074    0
	00000078    0
0000007c winedevice.exe
	00000080    0
	00000088    0
	0000008c    0
	00000090    0
00000094 plugplay.exe
	00000098    0
	000000a4    0
	000000a8    0
	000000ac    0
	000000d4    0
	000000d8    0
000000b0 winedevice.exe
	000000b4    0
	000000c8    0
	000000cc    0
	000000d0    0
	000000dc    0
000000e0 mscorsvw.exe
	000000e4    0
	000000ec    0
	000000f0    0
	000000f4    0
000000f8 rpcss.exe
	000000fc    0
	00000104    0
	00000108    0
	0000010c    0
	00000110    0
	00000114    0
	00000118    0
0000011c (D) Z:\home\clintwilson\LogosBible10\data\wine64_bottle\drive_c\users\clintwilson\AppData\Local\Logos\System\LogosIndexer.exe
	00000120    0 <==
	0000012c    0
System information:
    Wine build: wine-6.0.3 (Ubuntu 6.0.3~repack-1)
    Platform: x86_64
    Version: Windows 10
    Host system: Linux
    Host version: 6.9.3-76060903-generic`

**Comment by thw26** [comment]
Can you share long terminal output only after posting the contents to a new pastebin link? I'll need you to share the link as well.

In terms of finding the error, are you pulling that straight from the CLI or from the log file? If the CLI, check the log file at:

`~/.local/state/Logos_on_Linux/Logos_on_Linux.log`

It should be logged there. If it is not, try running the software with the `-D` flag which should log it.

**Comment by cwils1689** [comment]
Okay. Here's the pastebin link of the log file showing the error. The first code I share was from a Wine popup debug window.

[](https://pastebin.com/AUq6pWGY)

**Comment by cwils1689** [comment]
Okay. So, I noticed the log said, "Platform not supported: Windows 7 is the minimum supported version". So, I changed the Windows version in Winetricks for the Indexer from Vista to Windows 10. It is now running. It still crashed Logos, but I was able to reopen it and now both are running.

**Comment by eric-hansen** [comment]
What System 76 machine do you have?  I run a Darter Pro (newest model) and I'm curious if this is related to #156

**Comment by cwils1689** [comment]
It is the newest Darter Pro with the Ultra 7 CPU.

**Comment by cwils1689** [comment]
Switching the Indexer to Windows 10 has corrected this issue for me. But, I have found another issue. Logos crashes when I try to Print/Export a sermon from Sermon Builder. 
https://pastebin.com/xweXdHvj

**Comment by thw26** [comment]
Printing has been a known issue for us for a while. It requires us to do some `git bisect` work that none of the developers have had time for. We might perhaps be able to revisit this soon.

This was noted in the old wiki and that data has not yet been migrated over to the new repo. See: https://github.com/ferion11/LogosLinuxInstaller/wiki#known-issues

**Comment by thw26** [comment]
This PR has a commit to set the indexer to Win 10 by default:

https://github.com/FaithLife-Community/LogosLinuxInstaller/pull/164

**Comment by RoosZwart** [comment]
Using 4.0.0-alpha.14 on Ubuntu 24.04, I faced the same error as thw26. Indexer would not run, with the log reporting:

INFO: Platform not supported: Windows 7 is the minimum supported version
ERROR: Error running '/home/username/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage /home/username/LogosBible10/data/wine64_bottle/drive_c/users/username/AppData/Local/Logos/System/LogosIndexer.exe': 209

I was able to fix it in the same way: changed the Windows version in Winetricks for LogosIndexer.exe from Windows Vista to Windows 10, and the indexer ran without problems.
